sway-lib-core/src/ops.sw

```diff
@@ -621,32 +621,12 @@ fn rsh_with_carry(word: u64, shift_amount: u64) -> (u64, u64) {
     (shifted, carry)
 }
 
-/// Extract a single 64 bit word from a b256 value using the specified offset.
-fn get_word_from_b256(val: b256, offset: u64) -> u64 {
-    asm(r1: val, offset: offset, r2, res) {
-        add r2 r1 offset;
-        lw res r2 i0;
-        res: u64
-    }
-}
-
-/// Build a single b256 value from 4 64 bit words.
-fn compose(word_1: u64, word_2: u64, word_3: u64, word_4: u64) -> b256 {
-    let res: b256 = 0x0000000000000000000000000000000000000000000000000000000000000000;
-    asm(w1: word_1, w2: word_2, w3: word_3, w4: word_4, result: res) {
-        sw result w1 i0;
-        sw result w2 i1;
-        sw result w3 i2;
-        sw result w4 i3;
-        result: b256
-    }
+/// Build a single b256 value from a tuple of 4 u64 values.
+fn compose(words: (u64, u64, u64, u64)) -> b256 {
```

Comment:
I was worried for a second that this would not work if we call `compose` with literals as in `compose((0, 0, 0, 0))`, but I think that's fine given that `(0, 0, 0, 0)` is a reference type. However, will you please double check? :)

Reply:
At this stage we can assume the words are contiguous and in the same order in memory, but I'm not sure we should have an ASM block depending on that behaviour.  In this specific case it's so simple I think it's probably OK.